chain/chain/src/stateless_validation/chunk_validation.rs

```diff
@@ -412,13 +412,21 @@ pub fn pre_validate_chunk_state_witness(
                     }),
                     state_patch: Default::default(),
                 };
-
+                let config = runtime_adapter.get_runtime_config(protocol_version);
+                let validated_txs =
+                    ValidatedTransaction::new_list(&config, new_transactions.into_iter().cloned())
```

Comment:
That clone seem unnecessary since in the next step validated_tx are passed by reference (though I haven't looked beyond that). Is there any rust magic that we can do to improve that? For example can you introduce `new_list_ref` that keeps references instead of owned transactions? 

Feel free to push back if I'm being unreasonable :)

Reply:
This is a good catch.  If you follow the call chain down to [here](https://github.com/akhi3030/nearcore/blob/use-vt-in-pool/chain/pool/src/lib.rs#L313), we see that we pass references here to ultimately clone.  I'll try to fix this in this PR to see if this is a big change.

Reply:
Ended up fixing the issue in a separate PR and you got assigned as the reviewer there =).  https://github.com/near/nearcore/pull/13112